oak/server/rust/oak_runtime/src/runtime/mod.rs

```diff
@@ -530,7 +529,11 @@ impl Runtime {
                         }
                     }
                 }
-            })
+            });
+        if let Ok(Some(ReadStatus::Success(ref msg))) = result {
```

Comment:
If this needs to be done outside of the channel lock, please add a comment, otherwise the next person will just put it back where it was :)

Reply:
Done, added comment